packages/uix-core/src/tunnel/tunnel.ts

```diff
@@ -120,24 +122,47 @@ export class Tunnel extends EventEmitter {
       );
     }
 
-    const source = target.contentWindow;
     const config = Tunnel._normalizeConfig(options);
     const tunnel = new Tunnel(config);
     const messenger = new TunnelMessenger({
       myOrigin: window.location.origin,
-      targetOrigin: options.targetOrigin,
-      logger: options.logger || console,
+      targetOrigin: config.targetOrigin,
+      logger: config.logger,
     });
+    tunnel.on("destroyed", () =>
+      config.logger.log(
+        `Tunnel to iframe at ${config.targetOrigin} destroyed!`,
+        tunnel,
+        target
+      )
+    );
+    tunnel.on("connected", () =>
+      config.logger.log(
+        `Tunnel to iframe at ${config.targetOrigin} connected!`,
+        tunnel,
+        target
+      )
+    );
+    tunnel.on("error", (e) =>
+      config.logger.log(
+        `Tunnel to iframe at ${config.targetOrigin} error!`,
+        tunnel,
+        target,
+        e
+      )
+    );
     let frameStatusCheck: number;
     let timeout: number;
     const offerListener = (event: MessageEvent) => {
       if (
-        isFromOrigin(event, source, config.targetOrigin) &&
+        isFromOrigin(event, target.contentWindow, config.targetOrigin) &&
```

Comment:
Reassigning `target.contentWindow` was preserving old window objects that failed comparison.